tensorflow/core/protobuf/config.proto

```diff
@@ -388,9 +388,10 @@ message ConfigProto {
   // Optional list of all workers to use in this session.
   ClusterDef cluster_def = 14;
 
-  // If true, any resources such as Variables used in the session will not be
-  // shared with other sessions.
-  bool isolate_session_state = 15;
+  // This option is deprecated. The same effect can be achieved by setting
+  // share_session_state to an inverse value. For compatibility, this field
+  // still works if share_session_state is not turned on.
+  bool isolate_session_state = 15 [deprecated = true];
```

Comment:
If we deprecate this option, there seems to be no non-deprecated way to isolate session state when ClusterSpec propagation is disabled, which would mean that the new option is not a complete replacement.

Instead of deprecating this field, I would suggest adding a more specific option in the `Experimental` submessage below (e.g. something with a name like "share_session_state_in_clusterspec_propagation"), which will make it easier to merge this change promptly. After that has been merged for a while, the folks working on distributed APIs like `tf.distribute` can make a call about what the long-term supported configuration should be.

Reply:
@mrry Thanks for your suggestions!  
Agree with you that we need a long term API design to control isolation among sessions in distributed runtime.  
Codes are updated and is the pull request qualified now?